execute conda run with --no-capture-output

esc opened the pull request:
The default behaviour of `conda run` is to buffer the output and then
display the entire running output after the command has completed. For
long running compile or testing processes this is suboptimal from a
usability perspective since the progress can not be monitored in
real-time. Adding the flag `--no-capture-output` means that `conda run`
will not buffer the output but stream it directly to the terminal which
improves usability.

esc committed:
execute conda run with --no-capture-output

The default behaviour of `conda run` is to buffer the output and then
display the entire running output after the command has completed. For
long running compile or testing processes this is suboptimal from a
usability perspective since the progress can not be monitored in
real-time. Adding the flag `--no-capture-output` means that `conda run`
will not buffer the output but stream it directly to the terminal which
improves usability.

stuartarchibald commented on texasbbq.py:
```diff
@@ -265,7 +265,7 @@ def install(self, env):
                 conda_install(env, dep)
             os.chdir(self.name)
-            execute("conda run -n {} {}".format(env, self.install_command))
+            execute("conda run --no-capture-output -n {} {}".format(env, self.install_command))
```
Is a minimum version of `conda` now required for this?

esc replied:
yes, 4.9

stuartarchibald replied:
Would it be a good idea to enforce and/or document that?

esc replied:
I am not sure where to enforce that, the dependency management for `texasbbq` is the the responsibility  of the integration tests that use it. We can document it for sure.

esc replied:
Though it does download `miniconda` during operation, so I doubt this will be an issue, but for completeness, this is a good idea.

esc replied:
Implemented in d3a5ae99fc930024ef23f1d5c0435953840b5be8

esc committed:
document dependencies

As title.

stuartarchibald commented on README.rst:
```diff
@@ -47,10 +47,11 @@ module ``texasbbq`` will then provides a command-line interface for running the
-3.7 and has zero third-party dependencies. Hence it will probably run on a
-large variety of different CI systems and platforms. It provides a pure Python
-interface to ``miniconda`` without using a shell language.
+drive integration testing. This script will run on at least Python 2.7 and 3.7
```
Is there a python 2 version of conda 4.9?

esc replied:
Probably not, no, but that doesn't matter as Python 2.7 is only needed to bootstrap a miniconda which will include the correct version of `conda`.

esc replied:
Also, we always use `subprocess` to execute `conda` rather than importing it, which is another layer of isolation.

stuartarchibald replied:
ok, I don't think it's clear that it downloads its dependencies, xref https://github.com/numba/texasbbq/pull/22/files#diff-7b3ed02bc73dc06b7db906cf97aa91dec2b2eb21f2d92bc5caa761df5bbc168fR68

esc replied:
Hmmm, yes, this is somewhat unusual. Do you have any thoughts on how best to specify that?

esc replied:
Shall I perhaps add another sentence to the end of this paragraph to clarify that `miniconda` and `conda` will be downloaded?

esc replied:
I just realized, that I myself am and have been running this locally with python2.7:

```
 💣 zsh» python --version                                                                                                                                           :(
Python 2.7.16
 💣 zsh» python3 switchboard.py -t pygbm
::>> running: 'uname'
::>> stages are: '['miniconda', 'environment', 'install_source', 'install_target', 'tests']'
::>> targets are: '['pygbm']'
```

stuartarchibald replied:
> Shall I perhaps add another sentence to the end of this paragraph to clarify that `miniconda` and `conda` will be downloaded?

probably a good idea

esc replied:
done

esc committed:
Merge branch 'master' into conda_run_no_capture_output

* master:
  fix docstring
  CondaTarget does not have a checkout to cd into
  find CondaTarget subclasses also
  fix syntax and missing method
  initial implementation of a CondaTarget

esc committed:
add --no-capture-output to CondaTargt

esc commented:
Merged `master` and updated changes introduced by #23 -- ready for another round of reviews.

esc committed:
reword the paragrah

Ensure that we mention download of miniconda containing the right conda
version.

stuartarchibald commented on README.rst:
```diff
@@ -47,18 +47,25 @@ module ``texasbbq`` will then provides a command-line interface for running the
+probably run on a large variety of different CI systems and platforms. It
+provides a pure Python interface to ``conda`` without using a shell language.
+A self-contained miniconda distribution containing at least ``conda`` 4.9 will
+be dowwnloaded to ensure a clean build.
```
```suggestion
be downloaded to ensure a clean build.
```

esc committed:
Update README.rst

Co-authored-by: stuartarchibald <stuartarchibald@users.noreply.github.com>

stuartarchibald reviewed (approved):
Thanks for the patch and fixes.

esc commented:
@stuartarchibald thanks for the final review, have fixed the typo and will merge as soon as CI passes.
